### PR TITLE
Feature/clear on backspace

### PR DESCRIPTION
**Pull request by dana-c0914**
Fix for issue #861

### CODE REVIEW

File: src/ng-select/ng-select.component.ts [hunk 1]
```diff
@@ -785,7 +786,9 @@ export class NgSelectComponent implements OnDestroy, OnChanges, AfterViewInit, C
         }
 
         if (this.multiple) {
-            this.unselect(this.itemsList.lastSelectedItem)
+            if (this.clearOnBackspace) {
```
Comment (varnastadeus) [inline src/ng-select/ng-select.component.ts]: why this is applicable only for multiple? I think input should work for single select too

File: src/ng-select/ng-select.component.spec.ts [hunk 2]
```diff
@@ -1271,6 +1272,29 @@ describe('NgSelectComponent', function () {
...
+                fixture.componentInstance.multiple = true;
+                // fixture.componentInstance.clearOnBackspace = false;
+                select.clearOnBackspace = false;
+                fixture.componentInstance.cities = [...fixture.componentInstance.cities];
+                tickAndDetectChanges(fixture);
+                selectOption(fixture, KeyCode.ArrowDown, 1);
+                selectOption(fixture, KeyCode.ArrowDown, 1);
+                tickAndDetectChanges(fixture);
+                triggerKeyDownEvent(getNgSelectElement(fixture), KeyCode.Backspace);
+                const result: ObjectContaining<{value: any}>[] = [jasmine.objectContaining({
```
Comment (varnastadeus) [inline src/ng-select/ng-select.component.spec.ts]: asserting length should be enough

File: src/ng-select/ng-select.component.spec.ts [hunk 3]
```diff
@@ -1271,6 +1272,29 @@ describe('NgSelectComponent', function () {
                 expect(select.selectedItems).toEqual(result);
                 expect(remove).toHaveBeenCalled();
             }));
+
+            it('should not remove last selected value when multiple and clearOnBackspace false', fakeAsync(() => {
```
Comment (varnastadeus) [inline src/ng-select/ng-select.component.spec.ts]: cleanup test

File: src/ng-select/ng-select.component.ts (hunk 1 above)
Comment (dana-c0914) [reply src/ng-select/ng-select.component.ts]: at first I thought of only adding what needed in the issue, but yes sure should be applicable for single select too

**Comment by varnastadeus** [comment]
looks good, thanks for contributing!

**Pull request merged**